Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/Controllers/MultiPositionController.cs

```diff
@@ -0,0 +1,1001 @@
+                if (!string.IsNullOrEmpty(msg))
+                    Simulator.Confirmer.Information(msg);
+            }
+            messageDisplayed = true;
+        }
+
+        protected void CheckNeutralPosition()
+        {
+            bool setNext = false;
+            String previous = "";
+            foreach (Position pair in PositionsList)
+            {
+                if (setNext)
+                {
+                    currentPosition = pair.Type;
+                    Locomotive.SignalEvent(Common.Event.MPCChangePosition);
+                    break;
+                }
+                if (pair.Type == currentPosition)
+                {
+                    if (pair.Flag.ToLower() == "springloadedbackwards" || pair.Flag.ToLower() == "springloadedbackwardsimmediately")
+                    {
+                        setNext = true;
+                    }
+                    if (pair.Flag.ToLower() == "springloadedforwards" || pair.Flag.ToLower() == "springloadedforwardsimmediately")
+                    {
+                        currentPosition = previous;
+                        Locomotive.SignalEvent(Common.Event.MPCChangePosition);
+                        break;
+                    }
+                }
+                previous = pair.Type;
+            }
+        }
+
+        protected string GetPositionName(string type)
+        {
+            string ret = "";
+            foreach (Position p in PositionsList)
+            {
+                if (p.Type.ToLower() == type.ToLower())
+                    ret = p.Name;
+            }
+            return ret;
+        }
+
+        protected void EmergencyBrakes()
+        {
+            Locomotive.SetThrottlePercent(0);
+            Locomotive.SetDynamicBrakePercent(100);
+            Locomotive.TrainBrakeController.TCSEmergencyBraking = true;
+        }
+        public enum Movement
+        {
+            Forward,
+            Neutral,
+            Aft
+        };
+        public enum ControllerPosition
+        {
+            Neutral,
+            Drive,
+            ThrottleIncrease,
+            ThrottleDecrease,
+            ThrottleIncreaseFast,
+            ThrottleDecreaseFast,
+            DynamicBrakeIncrease, DynamicBrakeDecrease,
+            DynamicBrakeIncreaseFast,
+            TrainBrakeIncrease,
+            TrainBrakeDecrease,
+            EmergencyBrake,
+            ThrottleHold,
+            DynamicBrakeHold,
+            ThrottleIncreaseOrDynamicBrakeDecreaseFast,
+            ThrottleIncreaseOrDynamicBrakeDecrease,
+            DynamicBrakeIncreaseOrThrottleDecreaseFast,
+            DynamicBrakeIncreaseOrThrottleDecrease,
+            KeepCurrent,
+            SelectedSpeedIncrease,
+            SelectedSpeedDecrease,
+            SelectSpeedZero
+        };
+
+        public enum ControllerBinding
+        {
+            Throttle,
+            SelectedSpeed
+        }
+
+        public float GetDataOf(CabViewControl cvc)
```

Comment:
This is called by default in MSTSLocomotive.cs, but the effect is only for ORTS_MULTI_POSITION_CONTROLLER. Consider modifying the MSTSLocomotive GetDataOf so this is only called for case CABViewControlTypes.ORTS_MULTI_POSITION_CONTROLLER.

Reply:
Done